File: 2502-sort-the-people/sort-the-people.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
typedef struct {
    char* name;
    int height;
} Person;

int compare(const void* a, const void* b) {
    return ((Person*)b)->height - ((Person*)a)->height;
}

char** sortPeople(char** names, int namesSize, int* heights, int heightsSize, int* returnSize) {
    Person* people = (Person*)malloc(namesSize * sizeof(Person));
    
    for (int i = 0; i < namesSize; i++) {
        people[i].name = names[i];
        people[i].height = heights[i];
    }
    
    qsort(people, namesSize, sizeof(Person), compare);
    
    char** result = (char**)malloc(namesSize * sizeof(char*));
    for (int i = 0; i < namesSize; i++) {
        result[i] = people[i].name;
    }
    
    *returnSize = namesSize;
    free(people);
    return result;
}

File: 2502-sort-the-people/sort-the-people.java
import java.util.*;

class Solution {
    class Person implements Comparable<Person> {
        String name;
        int height;
        
        Person(String name, int height) {
            this.name = name;
            this.height = height;
        }
        
        @Override
        public int compareTo(Person other) {
            return other.height - this.height; // for descending order
        }
    }
    
    public String[] sortPeople(String[] names, int[] heights) {
        int n = names.length;
        Person[] people = new Person[n];
        
        for (int i = 0; i < n; i++) {
            people[i] = new Person(names[i], heights[i]);
        }
        
        Arrays.sort(people);
        
        String[] result = new String[n];
        for (int i = 0; i < n; i++) {
            result[i] = people[i].name;
        }
        
        return result;
    }
}